src/Microsoft.VisualStudio.Threading/JoinableTask.cs

```diff
@@ -941,7 +974,7 @@ internal void CompleteOnCurrentThread()
 
         internal void OnQueueCompleted()
         {
-            if (this.IsCompleted)
+            if (this.IsFullyCompleted)
```

Comment:
📝 Implementation code that used to reference `IsCompleted` now references `IsFullyCompleted`, except for the specific cases where `IsCompleted` is now used to provide a lock-free path.